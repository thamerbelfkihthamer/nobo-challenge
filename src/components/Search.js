import React, { Component } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import TextField from '@material-ui/core/TextField';
import IconButton from "@material-ui/core/IconButton";
import InputAdornment from "@material-ui/core/InputAdornment";
import SearchIcon from "@material-ui/icons/Search";

const classes = makeStyles(theme => ({
    icon: {
    marginRight: theme.spacing(2),
    },
    heroContent: {
    backgroundColor: theme.palette.background.paper,
    padding: theme.spacing(8, 0, 6),
    },
    heroButtons: {
    marginTop: theme.spacing(4),
    },
}));

export default class Search extends Component{

    constructor(props){
        super(props)
    }

    render(){
        const { searchTerm, handleSearch, onChange } = this.props;

        return (
            <div className={classes.heroContent} style={{ marginBottom: 100, marginTop: 50}}>
                <Container maxWidth="sm">

                    <form style={{ marginBottom: 50}} onSubmit={handleSearch}>
                        <TextField
                            value={searchTerm}
                            onChange={onChange}
                            id="outlined-full-width"
                            style={{ margin: 8 }}
                            placeholder="What's the tv show you are looking for ? "
                            helperText="ex : Fast Layne, Go Jetters ..."
                            fullWidth
                            margin="normal"
                            variant="outlined"
                            InputLabelProps={{
                            shrink: true,
                            }}
                            InputProps={{
                                endAdornment: (
                                <InputAdornment>
                                    <IconButton onClick={handleSearch}>
                                        <SearchIcon />
                                    </IconButton>
                                </InputAdornment>
                                )
                            }}
                        />
                    </form>
                </Container>
            </div>
        )
    }
}
