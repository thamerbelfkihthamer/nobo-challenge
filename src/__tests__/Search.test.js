import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Search from '../components/Search';
Enzyme.configure({ adapter: new Adapter() });

describe('Search', () => {

    it('render without crashing', () => {
        const dev = document.createElement('dev');
        ReactDOM.render(<Search />, dev);
        ReactDOM.unmountComponentAtNode(dev);
    });


    test('has a valid snapshot', () => {
        const search = renderer.create(<Search />);
        const tree = search.toJSON();
        expect(tree).toMatchSnapshot();
    });

})