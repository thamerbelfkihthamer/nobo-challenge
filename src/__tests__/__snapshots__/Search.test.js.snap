// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<div
  style={
    Object {
      "marginBottom": 100,
      "marginTop": 50,
    }
  }
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <form
      style={
        Object {
          "marginBottom": 50,
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          onClick={[Function]}
        >
          <input
            aria-describedby="outlined-full-width-helper-text"
            aria-invalid={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
            disabled={false}
            id="outlined-full-width"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="What's the tv show you are looking for ? "
            required={false}
            type="text"
          />
          <div
            className="MuiInputAdornment-root"
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-161 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-162"
              style={
                Object {
                  "width": 0,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
        <p
          className="MuiFormHelperText-root MuiFormHelperText-contained"
          id="outlined-full-width-helper-text"
        >
          ex : Fast Layne, Go Jetters ...
        </p>
      </div>
    </form>
  </div>
</div>
`;
